### PR TITLE
feat: [PIE-2963]: migrate to swc-loader

### PR DESCRIPTION
**Pull request by vkbansal**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: configs/webpack.common.js [hunk 1]
```diff
@@ -149,7 +146,7 @@ const config = {
     ]
   },
   resolve: {
-    extensions: ['.mjs', '.js', '.ts', '.tsx', '.json', '.ttf'],
+    extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx', '.json', '.ttf'],
```
Comment (knagurski) [inline configs/webpack.common.js:149]: Do we want to allow non-typescript React files? Not a problem, just curious

File: src/modules/33-auth-settings/pages/subscriptions/plans/planUtils.tsx [hunk 2]
```diff
@@ -169,7 +169,7 @@ export function getBtns({ isPlanDisabled, btnProps, getString }: GetBtnsProps):
     )
   })
 
-  return <Layout.Vertical spacing={'small'}>{...btns}</Layout.Vertical>
+  return <Layout.Vertical spacing={'small'}>{btns}</Layout.Vertical>
```
Comment (knagurski) [inline src/modules/33-auth-settings/pages/subscriptions/plans/planUtils.tsx:172]: Looks like some non-related stuff has crept into this PR. Try to keep your PRs focused on a single objective (getting SWC loader in)

**Review by kapilkumawat86 — APPROVED** [review]

File: configs/webpack.common.js (hunk 1 above)
Comment (vkbansal) [reply configs/webpack.common.js:149]: Devs from OPA and STO have added `jsx` code.

File: src/modules/33-auth-settings/pages/subscriptions/plans/planUtils.tsx (hunk 2 above)
Comment (vkbansal) [reply src/modules/33-auth-settings/pages/subscriptions/plans/planUtils.tsx:172]: This is required change. Else the build fails with `Spread children are not supported in React.`

**Review by knagurski — APPROVED** [review]
Great improvement, thanks 😊

**Review by abhinavrastogi-harness — APPROVED** [review]
looks good!

**Pull request merged**